config/initializers/rpush.rb

```diff
@@ -1,3 +1,16 @@
+# Utility to pull out notification type from the payload string
+module Rpush
+  class NotificationTypeParser
+    def self.parse(payload)
+      JSON.parse(payload).dig("data", "type") || "unknown"
+    rescue StandardError => e
+      # JSON parsing failed
+      HoneyBadger.notify(e)
+      "unknown"
+    end
+  end
+end
```

Comment:
What if we just inline this code in the block? We're also "patching" the `Rpush` module space which belongs to the gem, it's unlikely but if they add a `NotificationTypeParser` class, this code might break.

Since it's not reused, my vote goes for the inlining. After all `on.notification_delivered` down here is a regular Ruby block, you'll also have all the error handling in the same space

Reply:
Hmm, those are very good points @rhymes. I can see this code being re-used among different Rpush callbacks in the future, but right now it's not. I'll re-ingrate this inline.